code/modules/mob/living/carbon/carbon.dm

```diff
@@ -357,6 +359,22 @@
 /mob/living/carbon/proc/setDNA(var/datum/dna/newDNA)
 	dna = newDNA
 
+/mob/living/carbon/revive()
+	..()
+	if(handcuffed && !initial(handcuffed))
+		drop_specific_item(handcuffed)
+	handcuffed = initial(handcuffed)
+	update_handcuffed()
+
+	if(legcuffed && !initial(legcuffed))
+		unEquip(legcuffed)
+	legcuffed = initial(legcuffed)
+	update_inv_legcuffed()
+
+	if(reagents)
+		for(var/datum/reagent/R in reagents.reagent_list)
+			reagents.clear_reagents()
+		reagents.addiction_list.Cut()
```

Comment:
I'm honestly not sure. This could possibly have leaked over from my life refactor, I'll remove this block of code now.